crawl-ref/source/player.cc

```diff
@@ -7211,7 +7211,7 @@ bool player::corrode_equipment(const char* corrosion_source, int degree)
         if (!x_chance_in_y(prev_corr, prev_corr + 28))
         {
             props[CORROSION_KEY].get_int() += res_corr() ? 2 : 4;
```

Comment:
Would be a little simpler this way round:

```
prev_corr += res_corr() ? 2 : 4;
props[CORROSION_KEY] = prev_corr;
```

(Maybe rename `prev_corr` to just `corrosion`.  It doesn't do anything outside of this loop anyway.)